WordPress/wordpress-playground
# Shorten and simplify path to CORS proxy

## Motivation for the change, related issues

The CORS proxy URL is long and requires the user to remember a specific PHP file path.
```
https://wordpress-playground-cors-proxy.net/cors-proxy.php
```

Let's shorten the URL and let people forget the PHP file name. With this PR,
`https://wordpress-playground-cors-proxy.net/cors-proxy.php?https://example.com/`
can be replaced with
`https://wordpress-playground-cors-proxy.net/?https://example.com/`
or
`https://wordpress-playground-cors-proxy.net/https://example.com/`

## Implementation details

Instead of trying to derive the target URL by removing `$_SERVER['SCRIPT_NAME']` from `$_SERVER['REQUEST_URI']`, we switch to taking the target URL from `$_SERVER['PATH_INFO']` or `$_SERVER['QUERY_STRING']`.

In addition, we take advantage of WP Cloud's default script path and have that script define PATH_INFO if it doesn't exist. It shouldn't exist because WP Cloud isn't configured to provide path info today. (`$_SERVER['PATH_INFO']` === `/additional/info` when `$_SERVER['REQUEST_URI']` === '/path-to-script.php/additional/info').

## Testing Instructions (or ideally a Blueprint)

- Run unit tests from php-cors-proxy dir with `sh test.sh`
- Run CORS proxy deploy workflow and test manually with proxy site. The following proxy URLs should all work:
    - https://wordpress-playground-cors-proxy.net/cors-proxy.php?https://wordpress.org/
    - https://wordpress-playground-cors-proxy.net/?https://wordpress.org/
    - https://wordpress-playground-cors-proxy.net/https://wordpress.org/
